analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -233,12 +240,30 @@ class Cargo(
             { extractPackage(it, hashes) }
         )
 
-        val dependencies = resolveDependencyTree(projectName, projectVersion, metadata, packages) { id, devId ->
-            !isDevDependencyOf(id, devId, metadata)
-        }
+        val projectId = metadata["workspace_members"]
+            .map { it.textValueOrEmpty() }
+            .single { it.startsWith("$projectName $projectVersion") }
+
+        val projectNode = metadata["packages"].single { it["id"].textValueOrEmpty() == projectId }
+
+        fun filterDependencies(condition: (String) -> Boolean) =
+            projectNode["dependencies"].filter { node ->
+                val kind = node["kind"].textValueOrEmpty()
+                condition(kind)
+            }.mapNotNull {
+                val dependencyName = it["name"].textValue()
+                val version = getResolvedVersion(projectName, projectVersion, dependencyName, metadata)
+                if (version == null) null else Pair(dependencyName, version)
+            }
 
-        val devDependencies = resolveDependencyTree(projectName, projectVersion, metadata, packages) { id, devId ->
-            isDevDependencyOf(id, devId, metadata)
+        val directDependencies = filterDependencies { kind -> kind != "dev" && kind != "build" }
+        val directDevDependencies = filterDependencies { kind -> kind == "dev" || kind == "build" }
+
+        val dependencies = directDependencies.mapTo(sortedSetOf()) {
+            buildDependencyTree(name = it.first, version = it.second, packages = packages, metadata = metadata)
+        }
+        val devDependencies = directDevDependencies.mapTo(sortedSetOf()) {
```

Comment:
Those lines are identical as well.
Would it make sense to follow DRY here as well, maybe even append the mapping to above function ?

Reply:
Will do in a separate PR.